src/Shell/TinkerShell.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Laravel\Tinker\Shell;
+
+use Illuminate\Contracts\Events\Dispatcher;
+use Laravel\Tinker\Shell\Listener\TinkerEventEmitter;
+use Psy\Configuration;
+use Psy\Shell;
+
+class TinkerShell extends Shell
+{
+    /**
+     * @var Dispatcher
+     */
+    protected $dispatcher;
+
+    public function __construct(Dispatcher $dispatcher, Configuration $config = null)
```

Comment:
I think it's probably best that we inject the `TinkerEventEmitter` here instead. One less cascading dependency that way.